src/test/java/com/puppycrawl/tools/checkstyle/checks/OuterTypeFilenameCheckTest.java

```diff
@@ -165,4 +170,21 @@ public void testOuterTypeFilenameRecordsMethodRecordDef() throws Exception {
                 getNonCompilablePath("InputOuterTypeFilenameRecord.java"), expected);
     }
 
+    @Test
+    public void testStateIsClearedOnBeginTree1() throws Exception {
```

Comment:
I think it is already covered by https://github.com/checkstyle/checkstyle/commit/0169b0891563cd7e12d39aa42acc6e6b2cb0ee38

Could you close this pr if this is the case?

Reply:
Yes no issues